### PR TITLE
Heaven Renderer

### LINKED ISSUE #138: Copy, Delay, Hold, Pattern (Poly) process large input out of order

### PR DESCRIPTION
**Pull request by mat1jaczyyy**
New renderer for Apollo, actually pog
Fixes #138.

TODO Rendering:
* [x] Heaven Render Loop
* [x] Heaven Signal enqueuing
* [x] Screen snapshot system
* [x] Smart sleep/wake for Heaven
* [x] Handle Pro 78 message cap
* [x] CFY optimized output algorithm
* [x] Heaven job enqueuing
* [x] Stop jobs selectively (for choking)
* [x] Heaven sub-millisecond precision for jobs
* [x] Replace Fade Smoothness with FPS Limit

TODO Devices Rewrite:
* [x] Choke
* [x] Copy
* [x] Delay
* [x] Fade
* [x] Loop
* [x] Hold
* [x] Multi
* [x] Pattern
* [x] Pattern Editor
* [x] Preview

### CODE REVIEW

**Comment by mat1jaczyyy** [comment]
TESTING AND REVIEW TIME

**Comment by Brendonovich** [comment]
PogU

File: Apollo/Apollo.csproj [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <StartupObject>Apollo.Core.Program</StartupObject>
     <Version>1.0.0</Version>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <DefineConstants>PRERELEASE</DefineConstants>
```
Comment (mat1jaczyyy) [inline Apollo/Apollo.csproj:14]: Remind me to get rid of this before merging

**Comment by mat1jaczyyy** [comment]
here goes nothing

**Pull request merged**